ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+    '#options' => $options_hour,
+    '#default_value' => variable_get('ding_dibs_start_cron', 2),
+    '#required' => TRUE,
+    '#prefix' => $text . '<div class="container-inline">',
+  );
+  $form['ding_dibs_start_cron_minute'] = array(
+    '#type' => 'select',
+    '#options' => $options_minute,
+    '#default_value' => variable_get('ding_dibs_start_cron_minute', 0),
+    '#required' => TRUE,
+    '#suffix' => '</div><div class="description">' . t('Set the daily hh:ii to run this cron.') . '</div>',
+  );
+  $form['ding_dibs_emails'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Input emails that should be notified if something unusual happens with a payment.'),
+    '#description' => t('Enter up to ten email addresses Enter through the separator or comma, without blank lines.'),
+    '#default_value' => variable_get('ding_dibs_emails', ''),
+    '#required' => TRUE,
+    '#element_validate' => array('ding_dibs_emails_validate'),
+  );
+  return system_settings_form($form);
+}
+
+/**
+ * Validation forms element email report settings.
+ */
+function ding_dibs_emails_validate($element, &$form_state, $form) {
+  $arrays_email = preg_split("/\,|\r\n|\n/", $element['#value']);
+  if (count($arrays_email) < 10) {
+    foreach ($arrays_email as $email) {
+      $email = trim($email);
+      if (!empty($email) && !valid_email_address($email)) {
+        form_error($element, t('Please enter a valid email address.'));
+      }
+      elseif (empty($email)) {
+        form_error($element, t('Remove the empty string.'));
+      }
+    }
+  }
+  else {
+    form_error($element, t('Maximum 10 email addresses to send.'));
+  }
+}
+
+/**
+ * Callback builds a table of reports to email mailing.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_email_reports() {
+  $time = REQUEST_TIME - 10800;
+  $report = db_select('dibs_transactions', 'd')
+    ->fields('d', array(
+      'payment_order_id',
+      'payment_transaction_id',
+      'payment_time',
+      'customer_uid',
+      'order_info_short',
+      'order_info_long',
+      'payment_price',
+      'payment_status',
+      'provider_payment_status',
+      'dibs_capture_status',
+    ))
+    ->condition('payment_time', date('Y-m-d H:i:s', $time), '>')
+    ->condition('d.api_module', 'ding_dibs')
+    ->condition('d.api_delta', 1)
+    ->orderBy('payment_order_id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  $no_transactions_msg = '<p>' . t('No transactions found.') . '</p>';
+  if (!$max_count) {
+    return $no_transactions_msg;
+  }
+  for ($count = 0; $count < $max_count; $count++) {
+    if (!($report[$count]->payment_status == $report[$count]->provider_payment_status && $report[$count]->payment_status == $report[$count]->dibs_capture_status && $report[$count]->provider_payment_status == $report[$count]->dibs_capture_status)) {
+      $report_valid[] = $report[$count];
+    }
+  }
+  $rows = array();
```

Comment:
`$rows` is defined here but first used ~50 lines below. Move it close to where it is being used.